src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -274,36 +274,44 @@ private SNINpHandle CreateNpHandle(string fullServerName, long timerExpire, obje
         {
             if (parallel)
             {
-                SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                SNICommon.ReportSNIError(SNIProviders.NP_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
                 return null;
             }
 
-            // Named Pipe Format: np:\\<host name>\pipe\<pipe name>
-            const int minPipeLength = 10; // eg: //./pipe/a
-            string pipePath = fullServerName.Trim();
-            if (pipePath.Length < minPipeLength)
+            if(fullServerName.Length == 0 || fullServerName.IndexOf('/') != -1) // Pipe paths only allow back slashes
             {
                 SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.NP_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
                 return null;
             }
 
-            if (pipePath[0] != '\\' || pipePath[1] != '\\')
+            string serverName, pipeName;
+            if (fullServerName.IndexOf('\\') == -1)
             {
-                SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.NP_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
-                return null;
+                serverName = fullServerName;
+                pipeName = SNINpHandle.DefaultPipePath;
             }
-
-            int endofServerName = pipePath.IndexOf('\\', 2);
-
-            string pipeToken = @"\pipe\";
-            if (0 != string.Compare(pipeToken, 0, pipePath, endofServerName, pipeToken.Length))
+            else
             {
-                SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.NP_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
-                return null;
-            }
+                try
+                {
+                    Uri pipeURI = new Uri(fullServerName);
```

Comment:
The only way to get to this code path is by having an "np" prefix. See CreateConnectionHandle here: https://github.com/dotnet/corefx/blob/master/src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs#L188

The prefixes are stripped out to determine which protocol we need to use, with TCP as the default if no prefix is provided.